spec/unit/lib/cloud_controller/organization_quota_usage_spec.rb

```diff
@@ -0,0 +1,152 @@
+        allow(routing_api_client).to receive_messages(router_group: router_group, enabled?: true)
+        allow(router_group).to receive_messages(type: 'tcp', reservable_ports: [1234, 2, 2345, 3])
+      end
+
+      let!(:route) { Route.make(domain: domain, host: '', port: 1234, space: space1) }
+      let!(:route2) { Route.make(domain: domain, host: '', port: 2, space: space2) }
+      let!(:route3) { Route.make(domain: domain, host: '', port: 2345, space: space1) }
+      let!(:route4) { Route.make(domain: domain, host: '', port: 3, space: space3) }
+
+      it 'returns the number of ports for all spaces under the org' do
+        result_ports = org_usage.reserved_route_ports
+        expect(result_ports).to eq(3)
+      end
+    end
+
+    describe '#app_tasks' do
+      let!(:app) { VCAP::CloudController::AppModel.make(space: space1) }
+      let!(:app2) { VCAP::CloudController::AppModel.make(space: space1) }
+      let!(:app3) { VCAP::CloudController::AppModel.make(space: space2) }
+      let!(:app4) { VCAP::CloudController::AppModel.make(space: space3) }
+      let!(:droplet) do
+        VCAP::CloudController::DropletModel.make(
+          app_guid: app.guid,
+          state: VCAP::CloudController::DropletModel::STAGED_STATE
+        )
+      end
+
+      before do
+        VCAP::CloudController::FeatureFlag.make(name: 'task_creation', enabled: true, error_message: nil)
+        app.droplet = droplet
+        app.save
+      end
+
+      let!(:task) do
+        VCAP::CloudController::TaskModel.make(
+          name: 'task one',
+          command: 'echo task',
+          app_guid: app.guid,
+          droplet: app.droplet,
+          memory_in_mb: 5,
+          state: VCAP::CloudController::TaskModel::RUNNING_STATE
+        )
+      end
+      let!(:task2) do
+        VCAP::CloudController::TaskModel.make(
+          name: 'task two',
+          command: 'echo task',
+          app_guid: app2.guid,
+          droplet: app.droplet,
+          memory_in_mb: 5,
+          state: VCAP::CloudController::TaskModel::RUNNING_STATE
+        )
+      end
+      let!(:task3) do
+        VCAP::CloudController::TaskModel.make(
+          name: 'task false',
+          command: 'echo task',
+          app_guid: app3.guid,
+          droplet: app.droplet,
+          memory_in_mb: 5,
+          state: VCAP::CloudController::TaskModel::RUNNING_STATE
+        )
+      end
+      let!(:task4) do
+        VCAP::CloudController::TaskModel.make(
+          name: 'task three',
+          command: 'echo task',
+          app_guid: app4.guid,
+          droplet: app.droplet,
+          memory_in_mb: 5,
+          state: VCAP::CloudController::TaskModel::RUNNING_STATE
+        )
+      end
+
+      it 'returns the number of service keys for all spaces under the org' do
+        result_per_app_tasks = org_usage.app_tasks
+        expect(result_per_app_tasks).to eq(3)
+      end
+    end
+  end
+end
```

Comment:
I've changed the spec file and:
- used `let` and `before` instead of `let!`
- used `subject`
- tried to come up with a more  compact creation of test objects
- removed unnecessary mocks and parameters
- used `ManagedServiceInstance` (and `UserProvidedServiceInstance`)
- added tasks in `PENDING` and `CANCELING` state

```suggestion
require 'spec_helper'

module VCAP::CloudController
  RSpec.describe OrganizationQuotaUsage do
    let(:org) { Organization.make }
    let(:space1) { Space.make(organization: org) }
    let(:space2) { Space.make }
    let(:space3) { Space.make(organization: org) }

    subject(:org_usage) { OrganizationQuotaUsage.new(org) }

    describe '#routes' do
      before do
        [space1, space1, space2, space3].each { |s| Route.make(space: s) }
      end

      it 'returns the number of routes in all spaces under the org' do
        expect(org_usage.routes).to eq(3)
      end
    end

    describe '#service_instances' do
      before do
        [space1, space1, space2, space3].each { |s| ManagedServiceInstance.make(space: s) }
        UserProvidedServiceInstance.make(space: space1)
      end

      it 'returns the number of service instances in all spaces under the org' do
        expect(org_usage.service_instances).to eq(3)
      end
    end

    describe '#private_domains' do
      before do
        [space1, space1, space2, space3].each { |s| Domain.make(owning_organization: s.organization) }
      end

      it 'returns the number of private domains in all spaces under the org' do
        expect(org_usage.private_domains).to eq(3)
      end
    end

    describe '#service_keys' do
      before do
        [space1, space1, space2, space3].each { |s| ServiceKey.make(service_instance: ServiceInstance.make(space: s)) }
      end

      it 'returns the number of service keys in all spaces under the org' do
        expect(org_usage.service_keys).to eq(3)
      end
    end

    describe '#reserved_route_ports' do
      before do
        reservable_ports = [1234, 2, 2345, 3]
        router_group = instance_double(RoutingApi::RouterGroup, type: 'tcp', reservable_ports: reservable_ports)
        routing_api_client = instance_double(RoutingApi::Client, router_group: router_group, enabled?: true)
        allow(CloudController::DependencyLocator.instance).to receive(:routing_api_client).and_return(routing_api_client)
        domain = SharedDomain.make(router_group_guid: 'some-router-group')
        [space1, space1, space2, space3].each_with_index { |s, i| Route.make(space: s, domain: domain, host: '', port: reservable_ports[i]) }
      end

      it 'returns the number of reserved route ports in all spaces under the org' do
        expect(org_usage.reserved_route_ports).to eq(3)
      end
    end

    describe '#app_tasks' do
      before do
        [space1, space1, space2, space3].each { |s| TaskModel.make(app: AppModel.make(space: s), state: TaskModel::RUNNING_STATE) }
        TaskModel.make(app: AppModel.make(space: space1), state: TaskModel::PENDING_STATE)
        TaskModel.make(app: AppModel.make(space: space1), state: TaskModel::CANCELING_STATE)
      end

      it 'returns the number of app tasks in all spaces under the org' do
        expect(org_usage.app_tasks).to eq(4)
      end
    end
  end
end
```